dynamiq/nodes/agents/base.py

```diff
@@ -130,12 +142,19 @@ def execute(
         self.reset_run_state()
         config = ensure_config(config)
         self.run_on_node_execute_run(config.callbacks, **kwargs)
+        user_id = input_data.get("user_id", None)
+        self.memory.add_message(role=MessageRole.USER, content=input_data.get("input"), metadata={"user_id": user_id})
```

Comment:
There should be a different way to pass more metadata. Now it looks like we are only limited to `user_id`. What about `session_id` and other metadata that users might come up with?

Reply:
I would like to manage 'user_id' and 'session_id' independently while also retaining other metadata. I will provide the changes for your review.